packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -2209,28 +2230,17 @@ export class MergeTree {
     }
 
     // Assume called only when pos == len
-    private breakTie(
-        pos: number, len: number, node: IMergeNode, refSeq: number,
-        clientId: number, candidateSegment?: ISegment) {
+    private breakTie(pos: number, node: IMergeNode,  seq: number) {
         if (node.isLeaf()) {
             if (pos === 0) {
-                const segment = node;
-                const removalInfo = this.getRemovalInfo(segment);
-                if (removalInfo.removedSeq
-                    && removalInfo.removedSeq <= refSeq
-                    && removalInfo.removedSeq !== UnassignedSequenceNumber) {
-                    return false;
-                }
-
-                // Local change see everything
-                if (clientId === this.collabWindow.clientId) {
-                    return true;
-                }
-
-                if (node.seq !== UnassignedSequenceNumber) {
-                    // Ensure we merge right. newer segments should come before older segments
-                    return true;
-                }
+                // normalize the seq numbers
+                // if the new seg is local (UnassignedSequenceNumber) give it the highest possible
+                // seq for comparision, as it will get a seq higher than any other seq once sequences
+                // if the current seg is local (UnassignedSequenceNumber) give it the second highest
+                // possible seq, as the highest is reserved for the previous.
+                const newSeq = seq === UnassignedSequenceNumber ? Number.MAX_SAFE_INTEGER : seq;
+                const segSeq = node.seq === UnassignedSequenceNumber ? Number.MAX_SAFE_INTEGER - 1 : node.seq ?? 0;
+                return newSeq > segSeq;
```

Comment:
If it's equivalent, the original form is a little easier to read:
```ts
// If the new segment is locally pending, it always wins.
if (seq === UnassignedSequenceNumber) {
    return true;
}

// Otherwise if the existing segment is locally pending, it always win.
if (node.seq === UnasignedSequenceNumber) {
    return false;
}

// Otherwise the compare to see which segment is newer according to the global sequence.
return seq > node.seq;
```

Reply:
(...but FWIW, I've thought that it would have been convenient to define UnassignedSequenceNumber as Number.MAX_SAFE_INTEGER.)